### DISCUSSION TITLE
Compile ERROR ST32F103

### DISCUSSION THREAD

**Discussion by DatanIMU** [hathach/tinyusb#631] [Q&A]
**Set up**
[Mandatory] Provide details of your setup help us to reproduce the issue as quick as possible  
 - **PC OS**   : Ubuntu 18.04 
 - **Board**   : stm32f103bluepill
 - **Firmware**: examples/device/webusb_serial

**Describe the bug**
ace@ubuntu:~/tinyusb/examples/device/webusb_serial$ make BOARD=stm32f103bluepill all
CC system_stm32f1xx.o
In file included from /home/ace/tinyusb/lib/CMSIS_5/CMSIS/Core/Include/cmsis_compiler.h:54:0,
                 from /home/ace/tinyusb/lib/CMSIS_5/CMSIS/Core/Include/core_cm3.h:115,
                 from /home/ace/tinyusb/hw/mcu/st/cmsis_device_f1/Include/stm32f103xb.h:132,
                 from /home/ace/tinyusb/hw/mcu/st/cmsis_device_f1/Include/stm32f1xx.h:131,
                 from /home/ace/tinyusb/hw/mcu/st/cmsis_device_f1/Source/Templates/system_stm32f1xx.c:59:
/home/ace/tinyusb/lib/CMSIS_5/CMSIS/Core/Include/cmsis_gcc.h: In function '__cmsis_start':
/home/ace/tinyusb/lib/CMSIS_5/CMSIS/Core/Include/cmsis_gcc.h:151:3: error: 'for' loop initial declarations are only allowed in C99 or C11 mode
   for (__copy_table_t const* pTable = &__copy_table_start__; pTable < &__copy_table_end__; ++pTable) {

**To reproduce**
1. sudo apt-get install gcc-arm-none-eabi gdb-arm-none-eabi
2. make BOARD=stm32f103bluepill all

**Screenshots**
If applicable, add screenshots, bus capture to help explain your problem. 

**Log**
Please provide the stack's log (uart/rtt/swo) where the issue occurred, best with comments to explain the actual events. To enable logging, add `LOG=2` to to the make command if building with stock examples or set `CFG_TUSB_DEBUG=2` in your tusb_config.h. More information can be found at [example's readme](/docs/getting_started.md)

**Comment by hathach** [comment]
can you check and post your armgcc version here. Anyway, your armgcc is probably too old, try to install with xpacks 

```
npm install --global xpm
xpm install --global @xpack-dev-tools/arm-none-eabi-gcc@latest
```

installation path is either `$HOME/.local/xPacks/@xpack-dev-tools/arm-none-eabi-gcc/*/.content/bin`  or ``$HOME/opt/xPacks/@xpack-dev-tools/arm-none-eabi-gcc/*/.content/bin`  

or simply download it from arm
